File: CoreData/CoreData/Person.m
//
//  Person.m
//  CoreData
//
//  Created by shaofa on 14-5-4.
//  Copyright (c) 2014年 isoftstone. All rights reserved.
//

#import "Person.h"


@implementation Person

@dynamic name;
@dynamic school;
@dynamic age;
@dynamic cdid;

+ (Person *)createModel {
    return [NSEntityDescription insertNewObjectForEntityForName:@"Person" inManagedObjectContext:currentManagedObjectContext];
}

-(void)savePerson
{
    [currentManagedObjectContext save:nil];
}

-(Person *)findPersonWithKey:(NSString *)key
{
    if (key.length == 0) {
        return nil;
    }
    
    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Person" inManagedObjectContext:currentManagedObjectContext];
    

    
    NSFetchRequest *request = [[NSFetchRequest alloc] init];
    [request setEntity:entity];
    
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"self.cdid=%@" argumentArray:@[key]];
    [request setPredicate:predicate];
    
    NSArray *persons = [currentManagedObjectContext executeFetchRequest:request error:nil];
    
    return [persons lastObject];
}



@end

File: Swift-ClassDemo/Swift-ClassDemo/Person.swift
//
//  Person.swift
//  Swift-ClassDemo
//
//  Created by shaofa on 14-6-11.
//  Copyright (c) 2014年 isoftstone. All rights reserved.
//

import Foundation

class Person:NSObject,ChildDelegate{
    var _name: String?
    var _age:Int?
    var _id:String?
    
    init(name:String, age:Int, id:String){
        _name = name
        _age = age
        _id = id
    }
    
    func getAge() -> Int{
        return _age!
    }
    
    func getName() -> String{
        return _name!
    }
    func getID() -> String{
        return _id!
    }
    
    
    func feedChild(name:String!){
        println("\(_name)给\(name)喂奶")
    }
    
    func playWithChild(name:String!){
        println("\(_name)陪\(name)玩")
    }
    
    func takeShowerForChild(name:String!){
        println("\(_name)给\(name)洗澡")
    }
    
    func needFood(child:Child!)
    {
        self.feedChild(child._name)
    }
    
    func needClear(child:Child!)
    {
        self.takeShowerForChild(child._name)
    }
    
    func needFriend(child:Child!)
    {
        self.playWithChild(child._name)
    }
}

//func getAge() -> Int
//{
//    
//}
//func getName() -> String
//{
//
//}
//func getID() -> String
//{
//    
//}